src/sentry/db/models/base.py

```diff
@@ -67,39 +68,47 @@ def __hash__(self) -> int:
 
     def __reduce__(
         self,
-    ) -> Tuple[Callable[[int], models.Model], Tuple[Tuple[str, str]], Mapping[str, Any]]:
-        (model_unpickle, stuff, _) = super().__reduce__()
+    ) -> tuple[Callable[[int], models.Model], tuple[tuple[str, str]], Mapping[str, Any]]:
+        reduced = super().__reduce__()
+        assert isinstance(reduced, tuple), reduced
+        (model_unpickle, stuff, _) = reduced
         return (model_unpickle, stuff, self.__getstate__())
 
     def __setstate__(self, state: Mapping[str, Any]) -> None:
         self.__dict__.update(state)
 
+    def _get_relational_field(self, field_name: str) -> models.ForeignKey:
+        ret = self._meta.get_field(field_name)
+        if not isinstance(ret, models.ForeignKey):
```

Comment:
Is this a new check? Or would this cause an error before as well?

Reply:
the current code would crash with an `AttributeError` -- mypy helpfully pointed out that the current code was unsafe so I made the error message explicit